source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -45,8 +45,10 @@ Envoy::Ssl::CertificateDetailsPtr Utility::certificateDetails(X509* cert, const
       std::make_unique<envoy::admin::v3::CertificateDetails>();
   certificate_details->set_path(path);
   certificate_details->set_serial_number(Utility::getSerialNumberFromCertificate(*cert));
-  certificate_details->set_days_until_expiration(
-      Utility::getDaysUntilExpiration(cert, time_source));
+  const auto days_until_expiry = Utility::getDaysUntilExpiration(cert, time_source).has_value()
+                                     ? Utility::getDaysUntilExpiration(cert, time_source).value()
```

Comment:
Good catch, never consider this case before, I will update.

Reply:
Since I want to return `-1` but  Utility::getDaysUntilExpiration returns `absl::optional<size_t>`, can not use `value_or`, I introduce an intermediate variable here.